### ISSUE TITLE
Latest beta not unzipping ModernHttpClient

### ISSUE THREAD

**Issue by kentcb** [NuGet/Home#2518]
Using NuGet 3.4.2 beta instead of latest stable has resulted in my ModernHttpClient dependency failing to restore correctly:

![image](https://cloud.githubusercontent.com/assets/1901832/14414631/041b8b84-ffd5-11e5-9e08-5a9c45232ffb.png)

As you can see, it does not unzip the zip file. Reverting to 3.3.0.212 works fine.

**Comment by davidfowl** [comment]
Can you give some more details? Which NuGet client? What did you run?

**Comment by kentcb** [comment]
Command-line client, running on Mac/Mono as part of a build script. Annoyingly, it seems to work if I run directly from the CLI:

```
mono ../nuget.exe install modernhttpclient -outputdirectory Src/packages/
```

However, it definitely doesn't work using 3.4.2 in combination with FAKE 4.23.5. Using 3.3.0.212 with same FAKE  version _does_ work. My FAKE target looks like this:

```
Target "restore-packages" (fun _ ->
    !! "./**/packages.config"
    |> Seq.iter (
        RestorePackage (fun defaults ->
            {
                defaults with
                    OutputPath = (srcDir @@ "packages")
            })
        )
)
```

TBH, not sure if this is NuGet or FAKE bug now.

**Comment by davidfowl** [comment]
Sounds like a fake issue with the new nuget? If it works from the command line, not sure what else it could be..

**Comment by yishaigalatzer** [comment]
Thanks for reporting this issue. Can you please help us with unfolding what fake is doing differently from the command line?

**Comment by yishaigalatzer** [comment]
I do see one big difference between the two operations

In fake it looks like you would run `nuget.exe restore packages.config` and in the cli you run `mono nuget.exe install XXX`

**Comment by forki** [comment]
This is what FAKE does: https://github.com/fsharp/FAKE/blob/master/src/app/FakeLib/RestorePackageHelper.fs#L150 

It's calling "mono nuget.exe install [FullPath packages.config] -OutputDirectory [FullPath outpath] [sources]"

one quick check that you can do to see if it's the expected nuget tool:

```
!! "./**/packages.config"
|> Seq.iter (
    RestorePackage (fun defaults ->
        failwithf "%A" defaults.ToolPath
        {
            defaults with
                OutputPath = (srcDir @@ "packages")
        })
    )
```

In other words: it's not doing some magic here, but relative vs. absolute paths have always been some trouble for us in combination with nuget.exe (especially on mono)

**Comment by spadapet** [comment]
OK I can reproduce this on Linux with Mono on the command line.
So far I only repro with the ModernHttpClient package. I tried Newtonsoft.Json and it works. So maybe there is something messed up about the zipped files in ModernHttpClient.nupkg.
Or... there is something messed up about the mono implementation of System.IO.Compression. Don't know where the problem is yet.

**Comment by spadapet** [comment]
If I repackage ModernHttpClient then it works. The file times in the original ModernHttpClient.2.4.2 are bogus times in the future. Some code is failing to apply those files times to the extracted files on disk. I need to see if it's the NuGet code. That original package isn't good though.

**Comment by forki** [comment]
Ok that sounds indeed like a known mono zip bug. We ran into this in paket and built workarounds. _If_ this is the same bug then many packages have the same problem. Repackaging one package does not help here.

**Comment by yishaigalatzer** [comment]
Thanks @forki ! we have a fix in progress, if you have any feedback on the fix, please let us know.

**Comment by forki** [comment]
Yeah I guess we have a pretty similar fix. But unfortunately that's only the first of many mono zipping issues ;-(

**Comment by yishaigalatzer** [comment]
Did you file bugs on mono for these issues? Do you have links?

**Comment by forki** [comment]
Yes we filed everything. But it's hidden in closed paket issues since we
created workarounds or it got fixed in mono eventually. I just wanted you
to know about the situation so that you never trust that API...

**Comment by spadapet** [comment]
Fixed for 3.4.3

**Comment by migueldeicaza** [comment]
Which bugs are these System.IO.Compression bugs in Mono?

What I gather from other commits referenced here is that we throw an exception if we find an invalid date in the package, but such a bug has not been filed against Mono.

I could not find any bugs related to this in our bug tracking system.

Alternatively, can someone provide us with reproduction instructions?   I tried downloading NuGet 3.4.2, but it is no longer available for download from: https://dist.nuget.org/index.html   I also tried faking the urls without success.
